Added extension marker to enumerations

FabianEckermann opened the pull request:
Added extension marker to enumerations as defined in the standard (X.691)

eerimoq commented:
Please add unit tests for the new functionality, and I'll review the PR and then merge.

eerimoq commented on asn1tools/codecs/uper.py:
```diff
@@ -1044,6 +1058,14 @@ def compile_type(self, name, type_descriptor, module_name):
+        for value in values:
+            if value == '...':
+                extension = []
+        return extension
```
Does this function return anything but `None`? I mean, the loop iterates over the keys, which are integers. Comparing an integer to `'...'` will never be true. Or am I missing something?

FabianEckermann replied:
You are right, I fixed it

FabianEckermann commented:
I fixed the detection of extension markers for enumerations and updated the tests. However test_uper.py fails as the hardcoded encoded_message does not fit the encoded one as there are extension markers within test_lpp_14_3_0 that were ignored before

eerimoq commented:
Ok, update `test_lpp_14_3_0` so it passes with your changes. Also, add encode/decode testing in `test_encode_all_types` and `test_decode_all_types` of `Enumerated2` with value `two`.

FabianEckermann commented:
This should be a valid way to get the max key of the root element. I also added the tests you proposed.
